core/src/main/java/org/bitcoinj/core/Message.java

```diff
@@ -342,7 +346,18 @@ protected long readVarInt(int offset) throws ProtocolException {
             throw new ProtocolException(e);
         }
     }
-    
+
+    protected byte readByte() throws ProtocolException {
+        checkReadLength(1);
+        return payload[cursor++];
+    }
+
+    private void checkReadLength(int length) throws ProtocolException {
```

Comment:
This "extract method" refactoring could go into a separate commit, couldn't it?

Reply:
I just noticed you extracted the `checkReadLength()` method in order to add a  `readByte()` method. But what are you using this new method for? I don't see any references.